rimage/manifest.c

```diff
@@ -23,15 +23,28 @@
 #include <time.h>
 #include <sys/time.h>
 
-#include <uapi/user/manifest.h>
-
 #include "rimage.h"
 #include "file_format.h"
 #include "css.h"
 #include "cse.h"
 #include "plat_auth.h"
 #include "manifest.h"
 
+/**
+ * \brief Utility to get module pointer from position.
+ * \param [in,out] desc FW descriptor in manifest.
+ * \param [in] index Index of the module.
+ * \return Pointer to module descriptor.
+ *
+ * Note that index is not verified against OOB.
+ */
+static inline struct sof_man_module *sof_man_get_module(
+	struct sof_man_fw_desc *desc, int index)
+{
+	return (void *)desc + sizeof(struct sof_man_fw_header) +
+			index * sizeof(struct sof_man_module);
+}
+
```

Comment:
it's a bit sad to use duplicated code.
What about creating a 3rd directory like uapi/apps/ for stuff like manifest and trace?

Reply:
I know it's sad, but you pushed to have it solved atm. I wrote that we need some kind of directory like that, but I guess that's something that you and @lgirdwood  has to decide, so unless you figure out how to divide api/headers between apps/fw/kernel I made something like this.

Reply:
@lgirdwood any objection to create a 3rd directory, we'd have
ipc
user (can be renamed as kernel if needed)
apps

Reply:
or we could make this a macro
```
#define SOF_MAN_MODULE_OFFSET(index) \
        (sizeof(struct sof_man_fw_header) +         \
			index * sizeof(struct sof_man_module))
```
The pointer arithe is then done by the user and not in the header.

Reply:
@lgirdwood Your workaround is better if we want to just get rid of `static inline`s from uapi headers, but the real problem is that we want no logic in headers and now the question is if we consider simple addition and multiplication kind of logic?